src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -51,8 +51,11 @@ class TaskReplaceActor(
   // Old and new instances that have the Goal.Running
   val activeInstances = currentInstances.filter(_.state.goal == Goal.Running)
 
+  // Active instances that are healthy (or don't have health checks)
+  val healthyOrNoHealthCheckInstances = currentInstances.filter((i) => !i.hasConfiguredHealthChecks || i.state.healthy.getOrElse(false))
+
   // The ignition strategy for this run specification
-  private[this] val ignitionStrategy = computeRestartStrategy(runSpec, activeInstances.size)
+  private[this] val ignitionStrategy = computeRestartStrategy(runSpec, healthyOrNoHealthCheckInstances.size)
```

Comment:
for the method computeRestartStrategy lets change the name of the parameter from `runningInstancesCount` to `consideredHealthyCount`